### PR TITLE
Add error handling to composer GCF trigger sample

### PR DESCRIPTION
**Pull request by tswast**
Addressed feedback from bug 111401746.

Please review @TrevorEdwards

### CODE REVIEW

File: functions/composer-storage-trigger/test/index.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }),
+    mocks: {
+      fetch: FetchMock,
+      body: body,
+      bodyJson: bodyJson
+    }
+  };
+}
+
+test.cb.serial(`Does nothing for deleted files`, (t) => {
```
Comment (ace-n) [inline functions/composer-storage-trigger/test/index.test.js]: Can we remove `serial`?

File: functions/composer-storage-trigger/test/index.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  const event = {
+    data: {
+      file: `some-file`
+    }
+  };
+  const expectedMsg = `Something bad happened.`;
+  const sample = getSample();
+  sample.mocks.bodyJson.error = expectedMsg;
+
+  sample.program.triggerDag(event, (err, message) => {
```
Comment (ace-n) [inline functions/composer-storage-trigger/test/index.test.js]: Should we switch to promises at some point? (In a separate PR, of course.)

Comment (tswast) [reply functions/composer-storage-trigger/test/index.test.js]: I didn't realize we could return promises with GCF. Yeah, that'd be ideal.

**Comment by TrevorEdwards** [comment]
Thanks for making this. I think we should consider doing retry on failure in the future as well (maybe someone else can PR that?) We've had issues with execution_date timestamp collisions.

**Pull request merged**